### PR TITLE
KTOR-3516 Add auth from single provider if no WWW-Auth header is present

### PR DESCRIPTION
**Pull request by rsinukov**

### CODE REVIEW

File: ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/AuthProvider.kt [hunk 1]
```diff
@@ -24,14 +24,16 @@ public interface AuthProvider {
...
     public fun isApplicable(auth: HttpAuthHeader): Boolean
 
     /**
-     * Add authentication method headers and creds.
+     * Adds authentication method headers and credentials.
+     * @param authHeader value of `WWW-Authenticate` header from failed response, if exists
+     * @param request builder for an authenticated request
      */
-    public suspend fun addRequestHeaders(request: HttpRequestBuilder)
+    public suspend fun addRequestHeaders(authHeader: HttpAuthHeader?, request: HttpRequestBuilder)
```
Comment (e5l) [inline ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/AuthProvider.kt:36]: I would add the header as the last parameter with a default `null` value

**Review by e5l — APPROVED** [review]

**Comment by PatrykKleczkowski** [comment]
Hello @e5l 

Do you have any idea when it will be released?

Best regards!

**Comment by e5l** [comment]
Soon after kotlin 1.6.20!

**Comment by PatrykKleczkowski** [comment]
Thank You @e5l !
Is it related with this issue?
https://youtrack.jetbrains.com/issue/KTOR-2806

**Comment by e5l** [comment]
@rsinukov, could you take a look?

**Comment by rsinukov** [comment]
@PatrykKleczkowski Yes, it's the same issue

**Pull request merged**